crates/agglayer-config/src/lib.rs

```diff
@@ -48,24 +41,30 @@ pub use rate_limiting::RateLimitingConfig;
 pub use rpc::RpcConfig;
 
 /// The Agglayer configuration.
-#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[serde_with::serde_as]
+#[derive(Default, Deserialize, Serialize, Debug, PartialEq, Eq)]
 #[serde(rename_all = "kebab-case")]
+#[serde(deny_unknown_fields)]
+#[serde(default)]
 pub struct Config {
     /// A map of Zkevm node RPC endpoints for each rollup.
     ///
     /// The key is the rollup ID, and the value is the URL of the associated RPC
     /// endpoint.
-    #[serde(alias = "FullNodeRPCs", deserialize_with = "deserialize_rpc_map")]
+    #[serde_as(as = "HashMap<DisplayFromStr, _>")]
     pub full_node_rpcs: HashMap<u32, Url>,
```

Comment:
Note: There's also similar thing in rate limiter config for per-network rate limiting configuration but uses an ad-hoc implementation. This approach with `serde_as` is better.

Reply:
I'll not touch more than the scope of this PR, I created a ticket: #374 for that